crates/components-runtime/src/config.rs

```diff
@@ -0,0 +1,80 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct ComponentsConfiguration {
+    #[serde(default = "default_data_collection_components")]
+    pub data_collection: Vec<DataCollectionComponents>,
+    #[serde(default = "default_consent_mapping_components")]
+    pub consent_mapping: Vec<ConsentMappingComponents>,
+    pub cache: Option<PathBuf>,
+}
+
+fn default_data_collection_components() -> Vec<DataCollectionComponents> {
+    vec![]
+}
+
+fn default_consent_mapping_components() -> Vec<ConsentMappingComponents> {
+    vec![]
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct DataCollectionComponents {
+    pub name: String,
+    pub component: String,
+    pub credentials: HashMap<String, String>,
+    #[serde(default = "default_data_collection_component_config")]
+    pub config: DataCollectionComponentConfig,
```

Comment:
`Default` is already implemented on `DataCollectionComponentConfig` so no need for special functon

```suggestion
    #[serde(default)]
    pub config: DataCollectionComponentConfig,
```